toml/src/main/kotlin/org/rust/toml/resolve/CargoTomlFileReferenceProvider.kt

```diff
@@ -39,6 +40,10 @@ class CargoTomlFileReferenceProvider(private val patternType: PathPatternType) :
         }
 
         if (kind !is TomlLiteralKind.String) return emptyArray()
+
+        // Ignore paths with wildcards to avoid false positives, since they cannot be easily resolved
+        if (element.stringValue?.contains("*") == true) return emptyArray()
```

Comment:
Unfortunately, it doesn't work as expected because:
- in `"foo/*/"` path I expect that `foo` is resolved to the corresponding directory. Otherwise, it not only breaks "go to definition" action but also breaks refactoring like rename
- it allows `*` in any reference provided by `CargoTomlFileReferenceProvider` although they are allowed only in `workspace` table
- it doesn't cover other special symbols described in https://doc.rust-lang.org/cargo/reference/workspaces.html#the-workspace-section

Since the corresponding issue is not critical at all, we can provide a proper solution instead of quick hack to workaround the issue somehow. I suppose we can try to partially parse glob patterns and if the current segment or some previous one contains special symbols just return `null` in `com.intellij.psi.impl.source.resolve.reference.impl.providers.FileReferenceSet#createFileReference`.
I suppose `com.intellij.openapi.vcs.changes.ignore.cache.PatternCache#createRegex` may help here